### PR TITLE
Restructure torch.compile docs

### PR DESCRIPTION
**Pull request by svekars**
Current torch.compile docs have become a bit of a mess with the docs expanded in the left nav. This PR moves them under the torch.compiler menu item in the left nav. A bunch of rewrites were made in collaboration with @msaroufim to address formatting issues, latest updates that moved some of the APIs to the public torch.compiler namespace were addressed as well. The documentation is broken down in three categories that address three main audiences: PyTorch users, Pytorch Developers and PyTorch backend vendors. While, the user-facing documentation was significantly rewritten, dev docs and vendor docs kept mostly untouched. This can be addressed in the follow up PRs. 

cc @carljparker @albanD

### CODE REVIEW

**Comment by msaroufim** [comment]
Thank you @svekars! This was much needed, I tried clicking on the preview but got an error. Will recheck and click around in a bit

File: docs/source/torch.compiler.rst [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+   * - ``torch.compile(m, backend="onnxrt")``
+     - Uses ONNXRT for inference on CPU/GPU. `Read more <https://onnxruntime.ai/>`__
+   * - ``torch.compile(m, backend="tensorrt")``
+     - Uses ONNXRT to run TensorRT for inference optimizations. `Read more <https://github.com/onnx/onnx-tensorrt>`__
+   * - ``torch.compile(m, backend="ipex")``
+     - Uses IPEX for inference on CPU. `Read more <https://github.com/intel/intel-extension-for-pytorch>`__
+   * - ``torch.compile(m, backend="tvm")``
+     - Uses Apache TVM for inference optimizations. `Read more <https://tvm.apache.org/>`__
+
+Read More
```
Comment (msaroufim) [inline docs/source/torch.compiler.rst:76]: This is fantastic thank you

File: docs/source/torch.compiler.rst [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+.. note:: In some cases, the terms ``torch.compile``, TorchDynamo, ``torch.compiler``
+  might be used interchangeably in this documentation.
+
+As mentioned above, to run your workflows faster, ``torch.compile`` through
+TorchDynamo requires a backend that converts the captured graphs into fast
+machine code. Different backends can result in various optimization gains.
+The default backend is called TorchInductor, also known as *inductor*,
+TorchDynamo has a growing list of backends developed by our partners,
+which can be found in the `backends <https://github.com/pytorch/pytorch/tree/main/torch/_dynamo/backends>`_
```
Comment (msaroufim) [inline docs/source/torch.compiler.rst:40]: should be torch.compiler.list_backends

Also we actually don't want this list to grow XD so would rather we change verbiage to we're not accepting new backends but you can register them out of core following links for vendors

File: docs/source/torch.compiler_deepdive.rst [hunk 3]
```diff
@@ -1,7 +1,50 @@
...
+and check out the dev-discuss topics.
+
+   * `TorchDynamo deep-dive video <https://www.youtube.com/watch?v=egZB5Uxki0I>`__
+   * `dev-discuss topics <https://dev-discuss.pytorch.org/search?q=TorchDynamo%20order%3Alatest>`__
+
+TorchDynamo Internals
+~~~~~~~~~~~~~~~~~~~~~
 **Author**: `Jason Ansel <https://github.com/jansel>`_
 
+This section will go over some of the TorchDynamo internals and will
```
Comment (msaroufim) [inline docs/source/torch.compiler_deepdive.rst:45]: There was a user complaint that the guard section was missing what code snippet it was referring to, this was what the code looked like before then https://github.com/pytorch/torchdynamo/pull/1118/files

File: docs/source/torch.compiler_fine_grain_apis.rst [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+and the relevant use cases.
+
+The API that you can use to define portions of the code on which you can
+disable compilation are listed in the following table:
+
+.. csv-table:: TorchDynamo APIs to control fine-grained tracing
+   :header: "API", "Description", "When to use?"
+   :widths: auto
+
+   "``torch._dynamo.disable``", "Disables Dynamo on the decorated function as well as recursively invoked functions.", "Excellent for unblocking a user, if a small portion of the model cannot be handeled with ``torch.compile``."
```
Comment (msaroufim) [inline docs/source/torch.compiler_fine_grain_apis.rst:21]: a bunch of these are now available in torhc.compiler

File: docs/source/torch.compiler_get_started.rst [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+.. code-block:: python
+
+   import timm
+   import torch._dynamo as dynamo
+   import torch
+   model = timm.create_model('resnext101_32x8d', pretrained=True, num_classes=2)
+   opt_model = torch.compile(model, backend="inductor")
+   opt_model(torch.randn(64,3,7,7))
+
+Our goal with TorchDynamo and TorchInductor is to build the highest coverage
```
Comment (msaroufim) [inline docs/source/torch.compiler_get_started.rst:140]: can just remove this line

**Review by msaroufim — CHANGES_REQUESTED** [review]

File: docs/source/torch.compiler_deepdive.rst (hunk 3 above)
Comment (svekars) [reply docs/source/torch.compiler_deepdive.rst:45]: I guees it's about this:
> For example, the printouts for the first graph in the `toy_example`
   above are:

And there is no toy_example above. I guess this is the right one: https://github.com/pytorch/pytorch/blob/main/docs/source/compile/guards-overview.rst#L24

Comment (svekars) [reply docs/source/torch.compiler_deepdive.rst:45]: I guess the toy_example from here is missing. I'll copy it here.

**Review by msaroufim — APPROVED** [review]

**Comment by msaroufim** [comment]
very nice!

**Pull request closed without merging**